ERROR on CIS logs when deleting multiple VS CRD

#### Setup Details
 CIS Version : 2.X      
 Build: f5networks/k8s-bigip-ctlr:latest       
 BIGIP Version: Big IP x.x.x      
 AS3 Version: 3.x      
 Agent Mode: AS3/CCCL       
 Orchestration: K8S/OSCP      
 Orchestration Version:       
 Pool Mode: Cluster/Nodeport      
 Additional Setup details: <Platform/CNI Plugins/ cluster nodes/ etc>

#### Description  
 When deleting multiple VS you get the following ERROR on the CIS logs 
([ERROR] VirtualServer Not Found: default/xxxxxxx, failed to update VS status)

#### Steps To Reproduce
1)Create multiple VS and then delete them all "kubectl delete vs --all"

![image](https://user-images.githubusercontent.com/11005568/152474130-3ab6d71d-8282-4eb8-b5c3-d30f4936baeb.png)


2022/02/04 04:51:53 [DEBUG] [AS3] Response from BIG-IP: code: 200 --- tenant:cis --- message: success
2022/02/04 04:51:53 [ERROR] VirtualServer Not Found: default/tea-virtual-server, failed to update VS status
2022/02/04 04:51:53 [ERROR] VirtualServer Not Found: default/test-virtual-server, failed to update VS status

--- a/pkg/controller/backend.go
+++ b/pkg/controller/backend.go
@@ -139,9 +139,10 @@ func (agent *Agent) PostConfig(rsConfig ResourceConfigRequest) {
 		return
 	}
 	cfg := agentConfig{
-		data:      string(decl),
-		as3APIURL: agent.getAS3APIURL(nil),
-		id:        rsConfig.reqId,
+		data:            string(decl),
+		as3APIURL:       agent.getAS3APIURL(nil),
+		id:              rsConfig.reqId,
+		isDeleteRequest: rsConfig.isVSDeleted,
 	}
 
 	agent.Write(cfg)

--- a/pkg/controller/postManager.go
+++ b/pkg/controller/postManager.go
@@ -124,12 +124,13 @@ func (postMgr *PostManager) configWorker() {
 			respCfg, posted = postMgr.postOnEventOrTimeout(timeoutMedium, &cfg)
 		}
 
-		select {
-		case postMgr.respChan <- respCfg.id:
-		case <-postMgr.respChan:
-			postMgr.respChan <- respCfg.id
+		if !respCfg.isDeleteRequest {
+			select {
+			case postMgr.respChan <- respCfg.id:
+			case <-postMgr.respChan:
+				postMgr.respChan <- respCfg.id
+			}
 		}
-
 		firstPost = false
 	}
 }

--- a/pkg/controller/types.go
+++ b/pkg/controller/types.go
@@ -248,6 +248,7 @@ type (
 		dnsConfig          DNSConfig
 		defaultRouteDomain int
 		reqId              int
+		isVSDeleted        bool
 	}
 
 	// Pool config
@@ -503,9 +504,10 @@ type (
 	}
 
 	agentConfig struct {
-		data      string
-		as3APIURL string
-		id        int
+		data            string
+		as3APIURL       string
+		id              int
+		isDeleteRequest bool
 	}
 
 	globalSection struct {

--- a/pkg/controller/worker.go
+++ b/pkg/controller/worker.go
@@ -383,9 +383,12 @@ func (ctlr *Controller) processResource() bool {
 			shareNodes:         ctlr.shareNodes,
 			dnsConfig:          ctlr.resources.dnsConfig,
 			defaultRouteDomain: ctlr.defaultRouteDomain,
+			isVSDeleted:        rKey.rscDelete,
 		}
 		go ctlr.TeemData.PostTeemsData()
-		ctlr.enqueueReq(config)
+		if !config.isVSDeleted {
+			ctlr.enqueueReq(config)
+		}
 		ctlr.Agent.PostConfig(config)
 		ctlr.initState = false
 		ctlr.resources.updateOldConfig()

